### DISCUSSION TITLE
Suggestions for improving the new Admin console UI

### DISCUSSION THREAD

**Discussion by thomasdarimont** [keycloak/keycloak#11648] [Archive]
During a walk-through of the new Admin-Console with a customer we noticed a few things that could be improved in the new Admin-Console UI.

# Authentication Flow Configuration

Feature: Show Authentication Flow-Bindings in Flow list.

After creating a custom flow and binding it to a flow-type (e.g. Browser Flow)
Actual: ... the user does not see the Flow-Type of a particular flow.
Desired: ... the user should see the Flow-Type of a particular flow.

The flow type could be either shown as an additional "tag" or a column in the flows list.

# Provider information in  Server Info

Feature: Show Provider Information in Server Info / Realm Info.

The old admin console exposed the list of currently used providers in Server Info.
Customers used that information to determine whether certain extensions are available in a Keycloak instance, and potentially
additional information provided via `org.keycloak.provider.ServerInfoAwareProviderFactory`.

Currently the Server info of the new admin console does not show the list of installed providers.

# Client specific Mapper configuration
The old admin console allowed to configure client specific protocol mappers as an alternative to the implicit mapper configuration via Client scopes. Although the later provides a more generic configuration, support for specifying client specific mappers ease testing and fine grained configurations.

It would be useful to have that feature in the new admin console as well.

# Client secret should be masked in admin-ui

The old and new admin console show the client secret in plain text. This is problematic in demo / review situations.
It would be helpful if the client secet could be masked / hidden like the user password by default in the admin-console.
If administrators really need the plain text value they should be able to "reveal" or just copy the plain text value.

@jonkoops

**Comment by ssilvert** [comment]
Thanks for the feedback @thomasdarimont !

Could you make each of these into an issue at https://github.com/keycloak/keycloak-admin-ui/issues

Then we can get them prioritized and addressed.

**Reply by thomasdarimont** [reply]
Done @ssilvert 
- https://github.com/keycloak/keycloak-admin-ui/issues/2509
- https://github.com/keycloak/keycloak-admin-ui/issues/2510
- https://github.com/keycloak/keycloak-admin-ui/issues/2511
- https://github.com/keycloak/keycloak-admin-ui/issues/2512
- https://github.com/keycloak/keycloak-admin-ui/issues/2513

**Comment by cgeorgilakis** [comment]
At least 'Client specific Mapper configuration' is essential.
For service account clients, Keycloak autocreates three protocol mappers. In the new admin console, how can admin user view/modify them?

**Comment by ssilvert** [comment]
@cgeorgilakis We changed the way you add client-specific mapper.  See https://github.com/keycloak/keycloak-admin-ui/issues/2511

**Reply by cgeorgilakis** [reply]
Thanks
